src/test/librados/misc.cc

```diff
@@ -78,6 +78,7 @@ TEST(LibRadosMiscConnectFailure, ConnectFailure) {
       break;  // yay, we timed out
     // try again
     rados_shutdown(cluster);
+    ASSERT_EQ(0, rados_create(&cluster, NULL));
```

Comment:
right. `rados_shutdown()` will destroy the underlying `RadosClient`, and leave us an invalid pointer.